AWSIoT/AWSIoTDataManager.h

```diff
@@ -616,8 +620,13 @@ DEPRECATED_MSG_ATTRIBUTE("Use `updateUserMetaData` for updating the user meta da
 - (BOOL) publishData:(NSData *)data
              onTopic:(NSString *)topic
                  QoS:(AWSIoTMQTTQoS)qos
-         ackCallback:(AWSIoTMQTTAckBlock)ackCallback;
+         ackCallback:(nullable AWSIoTMQTTAckBlock)ackCallback;
```

Comment:
For this callback it must be nil depending on the value for QoS. An exception is thrown if the callback is defined when it is not allowed with that QoS value. This may be breaking change regardless since it changes the parameter to an optional.